angular-workspace/projects/ni/nimble-angular/src/directives/table-column/nimble-table-column.module.ts

```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { NimbleTableColumnTextDirective } from './text/nimble-table-column-text.directive';
+
+import '@ni/nimble-components/dist/esm/table-column/text';
+
+@NgModule({
+    declarations: [NimbleTableColumnTextDirective],
+    imports: [CommonModule],
+    exports: [NimbleTableColumnTextDirective]
+})
+export class NimbleTableColumnModule { }
```

Comment:
Is your vision that we'll have one module for all the Nimble-provided table columns? What's your rationale for this vs other options like:
1. add all columns to the table module
2. separate module per column type

To me the current approach seems like an undesirable middle ground; clients still have to make multiple module imports (unlike 1) and still may be importing extra column types that they don't use (unlike 2).

Reply:
Fair point, the concerns of which did cross my mind when I implemented, I just didn't have a strong opinion between the other options (and so I figured the middle-ground might have been better?). Do you or @rajsite have a sense for what the better choice is?

Reply:
I always have to re-learn how Angular tree shaking works but if it happens at the module level then 2. seems like a better approach to me. I'd rather clients deal with a bit of pain in having to import multiple modules than have end users potentially having to download extra column definitions that aren't needed.

If tree shaking can work to remove parts of a large module that are unused then 1 would be simpler.

Reply:
Because we import with side-effects (i.e. `import '@ni/nimble-components/dist/esm/table-column/text';`) it is not possible for dead-code elimination to run on the web component. It can't trace how symbols exported are used / not-used because there is not symbol to track, there is a global mutation to the CustomElementRegistry. So defining multiple imports with side-effects means they must always ship in every app. This is why each icon is defined in it's own NgModule.

So each web component import should happen in their own NgModule.

Reply:
We can probably mirror the component pattern
```
directive/table-column/base/(directive file assuming we need a spot to export any shared base class behaviors which we don't need yet, so we can skip for now)
directive/table-column/text/(directive and module file)
```

Reply:
Updated.